data/js/tomahawk.js

```diff
@@ -303,11 +305,12 @@ Tomahawk.Resolver = {
         return RSVP.Promise.resolve(this.testConfig(config)).then(function (results) {
             results = results || Tomahawk.ConfigTestResultType.Success;
             return results;
+        }, function (error) {
+            return error;
```

Comment:
Nope, I figured it was on purpose and I probably know what this is for. Just wanted to make sure this was your intention. I did this several times by accident when I actually meant to rethrow.